### ISSUE TITLE
Rule Change: Improve HTML id/name attribute parsing in no-missing-link-fragments

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#481]
### What rule do you want to change?

no-missing-link-fragments

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
<h1 id=bookmark>Bookmark</h1>
<h2 id = bookmark-2>Bookmark 2</h2>
<h3 id= bookmark-3>Bookmark 3</h3>
<h4 id =bookmark-4>Bookmark 4</h4>

[Link](#bookmark)
[Link](#bookmark-2)
[Link](#bookmark-3)
[Link](#bookmark-4)
```

### What does the rule currently do for this code?

The rule incorrectly reports missing fragments for links to headings with unquoted or spaced id/name attributes, even though the anchors exist in the HTML.

### What will the rule do after it's changed?

The rule will correctly recognize all valid id and name attributes, regardless of whether their values are quoted or unquoted, and regardless of spaces around the equals sign. No false positives will be reported for valid anchor links in these cases.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments'

HTML attributes can be quoted (single or double quotes) or unquoted, and the equals sign can have optional spaces before and after it.

_No response_

**Comment by lumirlumir** [comment]
Considering the comments, I am marking this PR as accepted, as there is a specification for this behavior.

<img width="1318" height="361" alt="Image" src="https://github.com/user-attachments/assets/d6e53c91-9d06-4f0e-9cd0-80226eabf9b8" />

<img width="690" height="39" alt="Image" src="https://github.com/user-attachments/assets/e31f9dbd-57f9-4dd5-b41f-e806db6e7577" />
